doc/design/networking/Investigation-on-NSX-integration.md

```diff
@@ -0,0 +1,180 @@
+  * `<vch name>-SG-network-a`    dynamic membership: NIC-3 (of containerVM3), NIC-4 (of containerVM4), NIC-52 (of containerVM5)
+  * `<vch name>-SG-network-b`    dynamic membership: NIC-5 (of containerVM5), NIC-6 (of containerVM6)
+
+We apply the following security policies to our security groups:
+  * `<vch name>-SP-0` is applied to all the security groups
+  * `<vch name>-SP-1` is applied to security groups `<vch name>-SG-bridge`, `<vch name>-SG-network-a` and `<vch name>-SG-network-b` 
+  * `<vch name>-SP-2` is applied to security groups `<vch name>-SG-bridge`, `<vch name>-SG-network-a` and `<vch name>-SG-network-b`
+  * `<vch name>-SP-3` is applied to security groups `<vch name>-SG-bridge`, `<vch name>-SG-network-a` and `<vch name>-SG-network-b`
+
+The result we get :
+  * all the networks are isolated to each other
+  * containerVMs in the network can connect to their gateway at VCH
+
+## NSX Manager API:
+![Code flow to use NSX Manager API] (pics/with-nsx.png)
+
+### API endpoint
+https://{nsxmanager's hostname or IP}/api 
+
+### API calls
+(details in http://pubs.vmware.com/nsx-63/topic/com.vmware.ICbase/PDF/nsx_63_api.pdf , note that many NSX API methods reference vCenter object IDs in URI parameters, query parameters, request bodies, and response bodies, also it takes XML as request body (sad) )
+* VCH create
+  * create security group 
+    * POST /2.0/services/securitygroup/bulk/{scopeId}
+  * create security policies and apply them to the security group
+    * POST /2.0/services/policy/securitypolicy (specify securityGroupBinding in the xml request body)
+
+* docker network create
+  * create security group `<vch name>-SG-<network name>`
+     * POST /2.0/services/securitygroup/bulk/{scopeId} (For the scopeId use globalroot-0 for non-universal security groups and universalroot-0 for universal security groups.)
+  * update security policies to apply them to the security group
+     * PUT /2.0/services/policy/securitypolicy/{ID}
+
+* docker network delete 
+  * check if any VMs attached to this security group
+     * GET /2.0/services/securitygroup/{objectId}/translation/virtualmachine 
+  * delete the network (security group, policy group)
+     * DELETE /2.0/services/securitytags/tag/{tagId}
+     * DELETE /2.0/services/securitygroup/{objectId} 
+
+* docker network list
+  * list all the security groups
+    * GET /2.0/services/securitygroup/internal/scope/{scopeId}
+  * filter security group names by `<vch name>-SG-` prefix
+
+* docker run --net=<network name> <image name>
+  * update security group `<vch name>-SG-<network name>` dynamic membership 
+    * PUT /2.0/services/securitygroup/bulk/{objectId}
+
+
+## Discussion:
+### Multi-tenants
+NSX-v supports multi-tenancy, and here is an example of a NSX network topology:
+
+![NSX-v multi-tenancy] (pics/nsx-multi-tenants.png)
+
+It worths to mention that we would expect only one VCH is created for one tenant. Otherwise there IP overlapping would issue. 
```

Comment:
We can replace this with something like:

"It is worth to mention that we would expect only one VCH to be created for one tenant. Otherwise IP overlapping would occur."